File: backend/src/main/java/com/backend/ord/services/impl/UserServiceImpl.java
package com.backend.ord.services.impl;

import com.backend.ord.domain.entities.User;
import com.backend.ord.domain.entities.UserSession;
import com.backend.ord.repositories.UserRepository;
import com.backend.ord.repositories.UserSessionRepository;
import com.backend.ord.services.UserService;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;
import java.util.UUID;
import java.util.stream.Collectors;
import java.util.stream.StreamSupport;

@Service
public class UserServiceImpl implements UserService {
    private final UserRepository userRepository;
    private final UserSessionRepository userSessionRepository;

    public UserServiceImpl(UserRepository userRepository, UserSessionRepository userSessionRepository) {
        this.userRepository = userRepository;
        this.userSessionRepository = userSessionRepository;
    }

    @Override
    public List<User> getAll() {
        Iterable<User> allUsers = userRepository.findAll();
        return StreamSupport.stream(allUsers.spliterator(), false).collect(Collectors.toList());
    }

    @Override
    public Optional<User> findById(UUID id) {
        return userRepository.findById(id);
    }

    @Override
    public User save(User user) {
        return userRepository.save(user);
    }

    @Override
    public Optional<User> findUserByAuthToken(String authToken) {
        return userSessionRepository.findByToken(authToken)
                .map(UserSession::getUser);
    }

    @Override
    public Optional<User> findUserByEmail(String email) {
        return userRepository.findByEmail(email);
    }
}

File: backend/src/main/kotlin/com/backend/ord/services/impl/UserServiceImpl.kt
package com.backend.ord.services.impl

import com.backend.ord.domain.persistance.entities.User
import com.backend.ord.repositories.UserRepository
import com.backend.ord.repositories.UserSessionRepository
import com.backend.ord.services.UserService
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service
import java.util.*

@Service
class UserServiceImpl(
    private val userRepository: UserRepository,
    private val userSessionRepository: UserSessionRepository
) : UserService {
    override fun findAll(): List<User> {
        return userRepository.findAll().toList().filterNotNull()
    }

    override fun findById(id: UUID): User? {
        return userRepository.findByIdOrNull(id)
    }

    override fun save(user: User): User {
        return userRepository.save(user)
    }

    override fun findUserByAuthToken(authToken: String): User? {
        return userSessionRepository.findByToken(authToken)?.user
    }

    override fun findUserByEmail(email: String): User? {
        return userRepository.findByEmail(email)
    }
}
